corehq/apps/userreports/views.py

```diff
@@ -1536,6 +1540,45 @@ def get_table(q):
         return export_response(Temp(path), params.format, adapter.display_name)
 
 
+@require_POST
+@api_auth()
+@require_permission(HqPermissions.view_reports)
+@toggles.SUPERSET_ANALYTICS.required_decorator()
+@api_throttle
+def subscribe_to_data_source_changes(request, domain, config_id):
+    from corehq.motech.models import ConnectionSettings
+    from corehq.motech.const import OAUTH2_CLIENT
+    from django.utils.translation import gettext as _
+
+    for param in ['webhook_url', 'client_id', 'client_secret', 'token_url', 'refresh_url']:
+        if param not in request.POST:
+            return HttpResponse(status=422, content=f"Missing parameter: {param}")
+
+    webhook_url = request.POST['webhook_url']
+    client_hostname = urlparse(webhook_url).hostname
+    connection_settings_name = f"{_('Connection')} - {client_hostname}"
+
+    conn_settings, _ = ConnectionSettings.objects.get_or_create(
```

Comment:
It looks like you're wanting to use [ConnectionSettings.objects.update_or_create()](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#update-or-create).

Also, don't forget to set the URL,
```python
        url=webhook_url,
```